packages/fetchai/contracts/erc1155/contract.py

```diff
@@ -88,20 +89,34 @@ def get_create_batch_transaction(
         :param gas: the gas to be used
         :return: the transaction object
         """
-        # create the transaction dict
-        nonce = ledger_api.api.eth.getTransactionCount(deployer_address)
-        instance = cls.get_instance(ledger_api, contract_address)
-        tx = instance.functions.createBatch(
-            deployer_address, token_ids
-        ).buildTransaction(
-            {
-                "gas": gas,
-                "gasPrice": ledger_api.api.toWei("50", "gwei"),
-                "nonce": nonce,
+        if ledger_api.identifier == "ethereum":
+            nonce = ledger_api.api.eth.getTransactionCount(deployer_address)
+            instance = cls.get_instance(ledger_api, contract_address)
+            tx = instance.functions.createBatch(
+                deployer_address, token_ids
+            ).buildTransaction(
+                {
+                    "gas": gas,
+                    "gasPrice": ledger_api.api.toWei("50", "gwei"),
+                    "nonce": nonce,
+                }
+            )
+            tx = cls._try_estimate_gas(ledger_api, tx)
+            return tx
+        elif ledger_api.identifier == "cosmos":
+            tokens = []
+            for token_id in token_ids:
+                tokens.append({"id": str(token_id), "path": str(token_id)})
+
+            msg = {
+                "create_batch": {"item_owner": str(deployer_address), "tokens": tokens}
             }
-        )
-        tx = cls._try_estimate_gas(ledger_api, tx)
-        return tx
+            tx = ledger_api.get_handle_transaction(
+                deployer_address, contract_address, msg, amount=0, tx_fee=0, gas=gas
```

Comment:
In conctract's get_handle_transaction is CosmosApi specific function and is missing in LedgerApi.
I can cast LedgerApi to CosmosApi - would have to import CosmosApi or I can add get_handle_transaction function to LedgerApi base class and Throw not implemented for other classes derived from LedgerApi.

Same thing applies to other CosmosApi functions called from contract -  try_execute_wasm_query, get_last_code_id, get_contract_address

Reply:
Please import CosmosApi and cast.